app/controllers/concerns/application_shared.rb

```diff
@@ -33,7 +33,7 @@ def set_taxonomy
           if params["#{taxonomy}_id"].blank? # the key is present and explicitly set to nil which indicates "any" context
             determined_taxonomy = nil
           else
-            determined_taxonomy = available.where(:id => params["#{taxonomy}_id"]).first
+            determined_taxonomy = available.friendly.find(params["#{taxonomy}_id"])
```

Comment:
this will raise `ActiveRecord::RecordNotFound` if it isn't found instead of returning `nil`

Reply:
good point, restored the behavior via rescue, IIRC using `friendly` required to use find